common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/config/YamlPropertySourceFactory.java

```diff
@@ -0,0 +1,28 @@
+package app.coronawarn.server.common.persistence.domain.config;
+
+import java.io.IOException;
+import java.util.Properties;
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.support.EncodedResource;
+import org.springframework.core.io.support.PropertySourceFactory;
+
+/**
+ * Because loading yaml files with @PropertySources is not supported in Spring,
+ * we need this custom implementation for processing the yamls and converting them
+ * to injectable properties in the Spring application context.
+ */
+public class YamlPropertySourceFactory implements PropertySourceFactory {
+
+  @Override
+  public PropertySource<?> createPropertySource(String name, EncodedResource encodedResource)
+      throws IOException {
```

Comment:
Please remove unthrown IOException present in method signature.

Reply:
Although this can be removed, the class is actually extends a framework interface which  is why it declares that exception. This will be called by the spring framework to deserialize the yaml and provide the injectable properties.